src/getMaxLevelDepthFromMapping.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Cube,
+  DataVisualizationWidgetMapping,
+  LevelCoordinates,
+  getHierarchy,
+  getLevel,
+} from "@activeviam/activeui-sdk";
+import { quote } from "@activeviam/mdx";
+
+/**
+ * Returns the column width for the tree table calculated by the baseWidth + (maxLevelDepth * levelMultiplier)
+ */
+export function getTreeColumnWidth({
+  maxLevelDepth,
+  mapping,
+  treeTableColumnWidth,
+  columnLevels,
+  cube,
+}: {
+  mapping: DataVisualizationWidgetMapping;
+  maxLevelDepth: number;
+  treeTableColumnWidth: [number, number];
+  columnLevels: LevelCoordinates[];
+  cube: Cube;
+}): { [columnKey: string]: number } {
+  let columnKey: string;
+
+  switch (mapping.rows[0].type) {
+    case "hierarchy": {
+      const hierarchy = columnLevels[0]
+        ? getHierarchy(
+            {
+              dimensionName: columnLevels[0].dimensionName,
+              hierarchyName: columnLevels[0].hierarchyName,
+            },
+            cube,
+          )
+        : undefined;
+
+      const firstLevelName = hierarchy && hierarchy.levels[1];
+      const firstLevel =
+        firstLevelName && columnLevels[0]
+          ? getLevel(
+              {
+                dimensionName: columnLevels[0].dimensionName,
+                hierarchyName: columnLevels[0].hierarchyName,
+                levelName: firstLevelName.name,
+              },
+              cube,
+            )
+          : undefined;
+
+      const { dimensionName, hierarchyName } = mapping.rows[0];
+
+      const levelName = firstLevel
+        ? firstLevel.name
+        : mapping.rows[0].levelName;
+
+      columnKey = quote(dimensionName, hierarchyName, levelName);
```

Comment:
This may be due to a potential bug on Atoti UI 5, but I found that if the first column is represented by a `regular` hierarchy (non-composite), and that hierarchy expanded down to a deeper level than the first, for example, the column is representing  `[Booking].[Desk].[BookId]`, its column key in the `columnWidths` property of the widget metadata  is represented by the first level of the hierarchy, in this case `[Booking].[Desk].[LegalEntity]`. To me this seems like a bug on the UI end since it would make more sense that the column should be represented by the more accurate `[Booking].[Desk].[BookId]`. This is to accommodate for that discrepancy.